src/test/javascript/spec/component/lecture/lecture-update.component.spec.ts

```diff
@@ -0,0 +1,336 @@
+    let lectureUpdateWizardComponent: LectureUpdateWizardComponent;
+
+    let lectureService: LectureService;
+    let lectureUpdateComponentFixture: ComponentFixture<LectureUpdateComponent>;
+    let lectureUpdateComponent: LectureUpdateComponent;
+    let router: Router;
+    let activatedRoute: ActivatedRoute;
+
+    let pastLecture: Lecture;
+
+    beforeEach(() => {
+        const yesterday = dayjs().subtract(1, 'day');
+
+        pastLecture = new Lecture();
+        pastLecture.id = 6;
+        pastLecture.title = 'past lecture';
+        pastLecture.endDate = yesterday;
+
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, FormsModule, MockModule(NgbTooltipModule)],
+            declarations: [
+                LectureUpdateComponent,
+                MockComponent(LectureUpdateWizardComponent),
+                MockComponent(FormDateTimePickerComponent),
+                MockComponent(MarkdownEditorComponent),
+                MockPipe(ArtemisTranslatePipe),
+                MockPipe(ArtemisDatePipe),
+                MockPipe(HtmlForMarkdownPipe),
+                MockRouterLinkDirective,
+            ],
+            providers: [
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: Router, useClass: MockRouter },
+                {
+                    provide: ActivatedRoute,
+                    useValue: {
+                        parent: {
+                            data: of({ course: { id: 1 } }),
+                        },
+                        queryParams: of({
+                            params: {},
+                        }),
+                        snapshot: {
+                            paramMap: convertToParamMap({
+                                courseId: '1',
+                            }),
+                        },
+                    },
+                },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                lectureUpdateComponentFixture = TestBed.createComponent(LectureUpdateComponent);
+                lectureUpdateComponent = lectureUpdateComponentFixture.componentInstance;
+
+                lectureUpdateWizardComponentFixture = TestBed.createComponent(LectureUpdateWizardComponent);
+                lectureUpdateWizardComponent = lectureUpdateWizardComponentFixture.componentInstance;
+
+                lectureService = TestBed.inject(LectureService);
+                router = TestBed.get(Router);
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should create lecture', fakeAsync(() => {
+        lectureUpdateComponent.lecture = { title: 'test1' } as Lecture;
+        const navigateSpy = jest.spyOn(router, 'navigate');
+
+        const createSpy = jest.spyOn(lectureService, 'create').mockReturnValue(
+            of<HttpResponse<Lecture>>(
```

Comment:
This can probably be inferred by Typescript, right?
```suggestion
            of(
```